src/autoqasm/types/types.py

```diff
@@ -104,8 +104,12 @@ def __init__(self, *args, annotations: str | Iterable[str] | None = None, **kwar
             raise errors.InvalidArrayDeclaration(
                 "Arrays may only be declared at the root scope of an AutoQASM main function."
             )
+
+        if not args:
+            raise errors.InvalidArrayDeclaration("init_expression must at least be provided.")
+        dimensions = [len(args[0])]
```

Comment:
We should keep support for multi-dimensional arrays as well. For example, today this works:
```py
@aq.main
def test():
    a = aq.ArrayVar([[1,2],[3,4]], dimensions=[2,2])
test.build().display()
```
and outputs:
```
OPENQASM 3.0;
array[int[32], 2, 2] a = {{1, 2}, {3, 4}};
```
Could you ensure this case works as well, and add a corresponding test?

Reply:
Thanks. I have added a test and ensured this still works.